qgym/benchmarks/metrics/initial_mapping_metrics.py

```diff
@@ -48,3 +71,72 @@ def __init__(
 
         Args:
         """
+
+
+class InitialMappingBenchmarker:
+
+    def __init__(
+        self,
+        generator: GraphGenerator | None = None,
+        *,
+        metrics: Iterable[InitialMappingMetric] | InitialMappingMetric,
+    ) -> None:
+        """Init of the :clas:`InitialMappingBenchmarker` class.
+
+        Args:
+            generator: Interaction graph generator to use during benchmarking
+            metrics: Metrics to compute.
+        """
+        self.generator = BasicGraphGenerator() if generator is None else generator
+        self.metrics = (
+            (metrics,) if isinstance(metrics, InitialMappingMetric) else tuple(metrics)
+        )
+        for metric in self.metrics:
+            if hasattr(metric, "connection_graph"):
+                connection_graph = deepcopy(metric.connection_graph)
+                self.generator.set_state_attributes(connection_graph=connection_graph)
+                break
+
+    def run(
+        self,
+        mapper: Mapper,
+        max_iter: int = 1000,
+        return_type: str = "raw",
+    ) -> NDArray[np.float64]:
+        """Run the benchmark.
+
+        Args:
+            mapper: Mapper to benchmark.
+            max_iter: Maximum number of iterations to benchmark.
+            return_type: Return type to return the results. Choose from ["raw",
+                "quartiles", "median", "mean"]
+
+        Returns:
+            NDArray containing the results from the benchmark.
+        """
+        return_type = check_string(return_type, "return_type", lower=True)
+
+        results: list[deque[float]] = [deque() for _ in self.metrics]
+        for i, interaction_graph in enumerate(self.generator, start=1):
+            mapping = mapper.compute_mapping(interaction_graph)
+            for metric, result_que in zip(self.metrics, results):
```

Comment:
How should I interpret the meaning of resultque?

Reply:
In the end I want to create a large array, which represents the following table
|               | graph 1 | graph2  | graph3  | ...  |
|----------|---|---|---|---|
| Metric 1 |  ... | ...  | ...  | ...  |
| Metric 2 |  ... | ...  | ...  | ...  |
| Metric 3 |  ... | ...  | ...  | ...  |

So I make a list of `deque` objects. Each `deque` is for one specific metric. I choose a `deque` since we need to do a lot of appending, which is exactly what the `deque` object is made for. Appending has a cost of $O(1)$ in the `deque`, where is has `O(n)` for a `list` and `ndarray`.